test/v5/createAmendForeignPropertyAnnualSubmission/def2/Def2_CreateAmendForeignPropertyAnnualSubmissionValidatorSpec.scala

```diff
@@ -0,0 +1,636 @@
+        val badValue = JsNumber(123.456)
+
+        "adjustment or allowances (except propertyIncomeAllowance) is invalid" when {
+
+          List(
+            (bodyWith(entry.update("/adjustments/privateUseAdjustment", badValue)), "/foreignProperty/0/adjustments/privateUseAdjustment"),
+            (bodyWith(entry.update("/adjustments/balancingCharge", badValue)), "/foreignProperty/0/adjustments/balancingCharge"),
+            (
+              bodyWith(entry.update("/allowances/annualInvestmentAllowance", badValue)),
+              "/foreignProperty/0/allowances/annualInvestmentAllowance"),
+            (
+              bodyWith(entry.update("/allowances/costOfReplacingDomesticItems", badValue)),
+              "/foreignProperty/0/allowances/costOfReplacingDomesticItems"),
+            (
+              bodyWith(entry.update("/allowances/zeroEmissionsGoodsVehicleAllowance", badValue)),
+              "/foreignProperty/0/allowances/zeroEmissionsGoodsVehicleAllowance"),
+            (bodyWith(entry.update("/allowances/otherCapitalAllowance", badValue)), "/foreignProperty/0/allowances/otherCapitalAllowance"),
+            (
+              bodyWith(entry.update("/allowances/electricChargePointAllowance", badValue)),
+              "/foreignProperty/0/allowances/electricChargePointAllowance"),
+            (
+              bodyWith(entry.update("/allowances/zeroEmissionsCarAllowance", badValue)),
+              "/foreignProperty/0/allowances/zeroEmissionsCarAllowance"),
+            (
+              bodyWith(entryWith("AFG", validStructuredBuildingAllowance.update("/amount", badValue))),
+              "/foreignProperty/0/allowances/structuredBuildingAllowance/0/amount"),
+            (
+              bodyWith(entryWith("AFG", validStructuredBuildingAllowance.update("/firstYear/qualifyingAmountExpenditure", badValue))),
+              "/foreignProperty/0/allowances/structuredBuildingAllowance/0/firstYear/qualifyingAmountExpenditure")
+          ).foreach((testValueFormatErrorWith _).tupled)
+        }
+
+        "propertyIncomeAllowance allowances is invalid" when {
+          List(
+            (
+              propertyIncomeAllowanceBodyWith(entryPropertyIncomeAllowance.update("/allowances/propertyIncomeAllowance", badValue)),
+              "/foreignProperty/0/allowances/propertyIncomeAllowance")
+          ).foreach(p => (testForPropertyIncomeAllowance _).tupled(p))
+        }
+
+        "propertyIncomeAllowance allowances is too large" when {
+          val bigValue = JsNumber(1000.01)
+          List(
+            (
+              propertyIncomeAllowanceBodyWith(entryPropertyIncomeAllowance.update("/allowances/propertyIncomeAllowance", bigValue)),
+              "/foreignProperty/0/allowances/propertyIncomeAllowance")
+          ).foreach(p => (testForPropertyIncomeAllowance _).tupled(p))
+        }
+
+        "passed a request body with empty fields except for additional (non-schema) properties" in {
```

Comment:
The name of this test doesn't  reflect what it does. Can you rename to (something like)
"passed a request body with multiple fields containing invalid values" since this is a single-error-multiple-paths case and move it down below the next closing brace.